util/chunk/compare.go

```diff
@@ -0,0 +1,167 @@
+		return cmpTime
+	case mysql.TypeDuration:
+		return cmpDuration
+	case mysql.TypeNewDecimal:
+		return cmpMyDecimal
+	case mysql.TypeSet, mysql.TypeEnum:
+		return cmpNameValue
+	case mysql.TypeBit:
+		return cmpBit
+	case mysql.TypeJSON:
+		return cmpJSON
+	}
+	return nil
+}
+
+func cmpNull(lNull, rNull bool) int {
+	if lNull && rNull {
+		return 0
+	}
+	if lNull {
+		return -1
+	}
+	return 1
+}
+
+func cmpInt64(l Row, lCol int, r Row, rCol int) int {
+	lNull, rNull := l.IsNull(lCol), r.IsNull(rCol)
+	if lNull || rNull {
+		return cmpNull(lNull, rNull)
+	}
+	return types.CompareInt64(l.GetInt64(lCol), r.GetInt64(rCol))
+}
+
+func cmpUint64(l Row, lCol int, r Row, rCol int) int {
+	lNull, rNull := l.IsNull(lCol), r.IsNull(rCol)
+	if lNull || rNull {
+		return cmpNull(lNull, rNull)
+	}
+	return types.CompareUint64(l.GetUint64(lCol), r.GetUint64(rCol))
+}
+
+func cmpString(l Row, lCol int, r Row, rCol int) int {
+	lNull, rNull := l.IsNull(lCol), r.IsNull(rCol)
+	if lNull || rNull {
+		return cmpNull(lNull, rNull)
+	}
+	return types.CompareString(l.GetString(lCol), r.GetString(rCol))
+}
+
+func cmpFloat32(l Row, lCol int, r Row, rCol int) int {
+	lNull, rNull := l.IsNull(lCol), r.IsNull(rCol)
+	if lNull || rNull {
+		return cmpNull(lNull, rNull)
+	}
+	return types.CompareFloat64(float64(l.GetFloat32(lCol)), float64(r.GetFloat32(rCol)))
+}
+
+func cmpFloat64(l Row, lCol int, r Row, rCol int) int {
+	lNull, rNull := l.IsNull(lCol), r.IsNull(rCol)
+	if lNull || rNull {
+		return cmpNull(lNull, rNull)
+	}
+	return types.CompareFloat64(l.GetFloat64(lCol), r.GetFloat64(rCol))
+}
+
+func cmpMyDecimal(l Row, lCol int, r Row, rCol int) int {
+	lNull, rNull := l.IsNull(lCol), r.IsNull(rCol)
+	if lNull || rNull {
+		return cmpNull(lNull, rNull)
+	}
+	lDec, rDec := l.GetMyDecimal(lCol), r.GetMyDecimal(rCol)
+	return lDec.Compare(rDec)
+}
+
+func cmpTime(l Row, lCol int, r Row, rCol int) int {
+	lNull, rNull := l.IsNull(lCol), r.IsNull(rCol)
+	if lNull || rNull {
+		return cmpNull(lNull, rNull)
+	}
+	lTime, rTime := l.GetTime(lCol), r.GetTime(rCol)
+	return lTime.Compare(rTime)
+}
+
+func cmpDuration(l Row, lCol int, r Row, rCol int) int {
+	lNull, rNull := l.IsNull(lCol), r.IsNull(rCol)
+	if lNull || rNull {
+		return cmpNull(lNull, rNull)
+	}
+	lDur, rDur := l.GetDuration(lCol), r.GetDuration(rCol)
+	return types.CompareInt64(int64(lDur.Duration), int64(rDur.Duration))
+}
+
+func cmpNameValue(l Row, lCol int, r Row, rCol int) int {
+	lNull, rNull := l.IsNull(lCol), r.IsNull(rCol)
+	if lNull || rNull {
+		return cmpNull(lNull, rNull)
+	}
+	_, lVal := l.getNameValue(lCol)
+	_, rVal := r.getNameValue(rCol)
+	return types.CompareUint64(lVal, rVal)
+}
+
+func cmpBit(l Row, lCol int, r Row, rCol int) int {
+	lNull, rNull := l.IsNull(lCol), r.IsNull(rCol)
+	if lNull || rNull {
+		return cmpNull(lNull, rNull)
+	}
+	lBit := types.BinaryLiteral(l.GetBytes(lCol))
+	rBit := types.BinaryLiteral(r.GetBytes(rCol))
+	lUint, err := lBit.ToInt()
+	terror.Log(err)
```

Comment:
What's the trace for?

Reply:
We log the error to pass `errcheck`, we never get this error or there is data corruption.

Reply:
Just for my personal culture, why simply not replace this: 
```
	lUint, err := lBit.ToInt()
	terror.Log(err)
```
by this:
```
	lUint, _ := lBit.ToInt()
```

Thank's in advance :wink: